src/code_80005FD0.c

```diff
@@ -561,22 +564,25 @@ s16 get_angle_between_waypoints(Vec3f arg0, Vec3f arg1) {
     return phi_v1;
 }
 
-s32 func_80006018(f32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5, f32 arg6, f32 arg7) {
-    f32 temp_f0;
+bool is_collide_with_vehicle(f32 vehicleX, f32 vehicleY, f32 vehicleVelocityX, f32 vehicleVelocityZ, f32 distanceX,
+                             f32 distanceY, f32 playerX, f32 playerZ) {
+    f32 velocity;
     f32 temp_f18;
 
-    temp_f0 = sqrtf((arg2 * arg2) + (arg3 * arg3));
-    if (temp_f0 < 0.01f) {
-        return 0;
+    velocity = sqrtf((vehicleVelocityX * vehicleVelocityX) + (vehicleVelocityZ * vehicleVelocityZ));
+    if (velocity < 0.01f) {
+        return false;
     }
-    temp_f18 = ((arg2 / temp_f0) * (arg6 - arg0)) + ((arg3 / temp_f0) * (arg7 - arg1));
-    if ((-arg4 < temp_f18) && (temp_f18 < arg4)) {
-        temp_f18 = ((arg3 / temp_f0) * (arg6 - arg0)) + (-(arg2 / temp_f0) * (arg7 - arg1));
-        if ((-arg5 < temp_f18) && (temp_f18 < arg5)) {
-            return 1;
+    temp_f18 =
```

Comment:
I don't remember how far I did documentation in collision.c

But this function looks similar to some of them from the first half that are similar to this.